pkg/kubernetes/deployment/extnv1beta1/v1alpha1/deployment.go

```diff
@@ -79,17 +89,134 @@ func (d *Deploy) GoString() string {
 func NewBuilder() *Builder {
 	return &Builder{
 		deployment: &Deploy{
-			object: &extnv1beta1.Deployment{},
+			Object: &extnv1beta1.Deployment{},
 		},
 	}
 }
 
+func (b *Builder) WithName(name string) *Builder {
+	if len(name) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: missing name"),
+		)
+		return b
+	}
+	b.deployment.Object.Name = name
+	return b
+}
+
+func (b *Builder) WithNameSpace(ns string) *Builder {
+	if len(ns) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: missing namespace"),
+		)
+		return b
+	}
+	b.deployment.Object.Namespace = ns
+	return b
+}
+
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: missing labels"),
+		)
+		return b
+	}
+	b.deployment.Object.Labels = labels
+	return b
+}
+
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errors = append(
+			b.errors,
+			errors.New("failed to build deployment object: missing annotations"),
+		)
+		return b
+	}
+	b.deployment.Object.Annotations = annotations
+	return b
+}
+
+func (b *Builder) WithOwnerReferences(csp *apis.NewTestCStorPool) *Builder {
```

Comment:
this becomes problem if other deployments want to add different references..
let deployment builder WithOwnerReferences have argument as metav1.OwnerReference

Reply:
Yes -- make sense. Will do it.